src/Microsoft.ML.StandardLearners/FactorizationMachine/FieldAwareFactorizationMachineModelParameters.cs

```diff
@@ -15,21 +15,21 @@
 using Microsoft.ML.Runtime.Internal.Utilities;
 using Microsoft.ML.Runtime.Model;
 
-[assembly: LoadableClass(typeof(FieldAwareFactorizationMachinePredictor), null, typeof(SignatureLoadModel), "Field Aware Factorization Machine", FieldAwareFactorizationMachinePredictor.LoaderSignature)]
+[assembly: LoadableClass(typeof(FieldAwareFactorizationMachineModelParameters), null, typeof(SignatureLoadModel), "Field Aware Factorization Machine", FieldAwareFactorizationMachineModelParameters.LoaderSignature)]
 
 [assembly: LoadableClass(typeof(FieldAwareFactorizationMachinePredictionTransformer), typeof(FieldAwareFactorizationMachinePredictionTransformer), null, typeof(SignatureLoadModel),
     "", FieldAwareFactorizationMachinePredictionTransformer.LoaderSignature)]
 
 namespace Microsoft.ML.Runtime.FactorizationMachine
 {
-    public sealed class FieldAwareFactorizationMachinePredictor : PredictorBase<float>, ISchemaBindableMapper, ICanSaveModel
+    public sealed class FieldAwareFactorizationMachineModelParameters : ModelParametersBase<float>, ISchemaBindableMapper
     {
-        public const string LoaderSignature = "FieldAwareFactMacPredict";
+        internal const string LoaderSignature = "FieldAwareFactMacPredict";
         public override PredictionKind PredictionKind => PredictionKind.BinaryClassification;
         private bool _norm;
-        internal int FieldCount { get; }
-        internal int FeatureCount { get; }
-        internal int LatentDim { get; }
+        public int FieldCount { get; }
+        public int FeatureCount { get; }
+        public int LatentDim { get; }
         internal int LatentDimAligned { get; }
         private readonly float[] _linearWeights;
```

Comment:
How you decide which one become public and which are not?
I wouldn't mind to get my hands on Weights, (as long as they are immutableArray)